plugins/aws/cli_provisioner.go

```diff
@@ -0,0 +1,58 @@
+package aws
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/1Password/shell-plugins/sdk"
+)
+
+const defaultProfileName = "default"
+
+type awsCLIProvisioner struct {
+}
+
+func NewAwsCLIProvisioner() sdk.Provisioner {
+	return awsCLIProvisioner{}
+}
+
+func (p awsCLIProvisioner) Provision(ctx context.Context, in sdk.ProvisionInput, out *sdk.ProvisionOutput) {
+	profileName := defaultProfileName
+
+	if profile := stripAndReturnProfileFlag(out); profile != nil {
+		profileName = *profile
+	} else if profile := os.Getenv("AWS_PROFILE"); profile != "" {
+		profileName = profile
+	}
+	NewSTSProvisioner(profileName).Provision(ctx, in, out)
```

Comment:
We'll need it to support other tools that use AWS (SDK), like Terraform. We don't want to look for the `--profile` flag in those tools.

Reply:
Ah, okay, that explains why `NewSTSProvisioner()` exists :-) In that case it becomes more of a nitpick, but it'd still be nicer to not construct a full provisioner if we are only interested in its `Provision()` method.

Reply:
I get it that line 28 looks weird for now, but the `STSProvisioner` is meant to be used in other parts of the codebase as well, pretty soon, in places where the contained functionality will need to be passed specifically as a provisioner.

To be more specific, in the case of terraform, when the terraform shell plugin knows it needs to provision AWS, it will go directly for the `STSProvisioner` and not the `AwsCliProvisioner`. And for re-usability, we have have `AwsCliProvisioner` rely on `StsProvisioner` logic. Does this make sense? Do you think there could be another way of doing it?

Reply:
Actually I changed the code now so it makes use of the provisioners as I explained.